reactive-streams/src/main/scala/fs2/interop/reactivestreams/StreamSubscriber.scala

```diff
@@ -69,15 +69,19 @@ final class StreamSubscriber[F[_], A](
     sub.onError(t)
   }
 
-  def stream(subscribe: F[Unit]): Stream[F, A] = sub.stream(subscribe)
+  def streamChunk(subscribe: F[Unit]): Stream[F, Chunk[A]] = sub.stream(subscribe)
+  def stream(subscribe: F[Unit]): Stream[F, A] = streamChunk(subscribe).unchunks
 
   private def nonNull[B](b: B): Unit = if (b == null) throw new NullPointerException()
 }
 
 object StreamSubscriber {
 
+  def apply[F[_]: Async, A](bufferSize: Long): F[StreamSubscriber[F, A]] =
+    fsm[F, A](bufferSize).map(new StreamSubscriber(_))
+
   def apply[F[_]: Async, A]: F[StreamSubscriber[F, A]] =
-    fsm[F, A].map(new StreamSubscriber(_))
+    apply(bufferSize = 10L)
```

Comment:
This changes the previous behavior asking elements one by one.
It's also a magic number coming from a quick thinking about the trade-offs memory vs performance. 
Should we handle that differently? Using `1L` to keep the previous behavior? Using some configuration?

Reply:
What's the behavior if requesting 10 but there's only 5 available?

This does feel a bit too magical IMO -- I lean towards setting it to 1 or an arbitrary power of 2 :)

Reply:
It only 5 are available, then we build a chunk of 5 elements, and it's being send in the `OnComplete` step: https://github.com/typelevel/fs2/pull/2666/files#diff-493b2533a887032dfbf8ea09dca7201413e124b5f45732ad6a877b8d7f69ba2cR178-R180

By setting it to `1` by default can lead to bad performances by default. I'd suggest to take an arbitrary number, not too high to avoid buffering too many elements in memory. `16`?

Reply:
https://github.com/typelevel/fs2/pull/2666/commits/5de1aa4e0bbcb109a875900a0b9b6ee55d424b4b

Reply:
I would strongly recommend *not* changing existing behavior in source-compatible ways

Reply:
Yep agreed, let's stick with 1 here as that's what every has used for years.

Reply:
1 versus 10 versus 16 could make a performance difference in certain types of streams but not others, where folks would need 100 or 1000 or more. Hence, I think it's best if we make no assumptions at all. So let's go with deprecating the old API and forcing the buffer size decision on to users.

Reply:
https://github.com/typelevel/fs2/pull/2666/commits/4ad2bbb6ac4ef7cb44d64f960cacbf3ff70627b0
to keep the existing behavior + deprecations